src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -215,8 +216,8 @@ export class Kernel implements IKernel {
                     try {
                         traceInfo(`Starting Notebook in kernel.ts id = ${this.kernelConnectionMetadata.id}`);
                         this.notebook = await this.notebookProvider.getOrCreateNotebook({
-                            identity: this.uri,
-                            resource: this.uri,
+                            identity: this.notebookUri,
+                            resource: this.resourceUri,
```

Comment:
Now, we're passing the right `resource`, else we'd be passing the Uri of the Interactive Notebook Uri instead of the the Python file Uri (i.e. works the way it did with inteactive webview)